corehq/util/view_utils.py

```diff
@@ -172,7 +176,7 @@ def request_as_dict(request):
         ),
         datespan=request.datespan,
         couch_user=None,
-        can_access_all_locations=request.can_access_all_locations
+        can_access_all_locations=can_access_all_locations
```

Comment:
It wasn't, that was the issue. I suppose there is a fair amount of our code that presumes the existence of Middleware -- i.e. our code mostly shuts down if `UsersMiddleware` isn't setting the domain on requests, but I think, ideally, middleware should be optional (like you can enable logging middleware or not, but it won't crash your app). While I think location access middleware will always be enabled for prod, I made this change because there was a reasonable default for it being off, and it made testing views much easier.

Reply:
I think it's a case-by-case basis of whether a given middleware is expected to be present or not. Middleware is definitely the easiest place to implement things like setting `request.domain`, so I think it makes sense to do that in middleware. It is also true that much of our view-level code depends on `request.domain`, and therefore it is fine to say `UsersMiddleware` is required in HQ and therefore fine for view code to assume it is present.

On the other hand, any middleware that is expected to be optional, meaning may be legitimately excluded in an environment, then any code that assumes it is present has a bug.

It looks like [`LocationAccessMiddleware` is included in `settings.MIDDLEWARE` by default](https://github.com/dimagi/commcare-hq/blob/88d3fe783b6e4f38b04dea7bc3f777c6b71a7280/settings.py#L168), and the normal way for extra middleware to be added is by setting `localsettings.LOCAL_MIDDLEWARE`. Curious how you concluded that the reasonable default is for it to be off?

Reply:
I definitely agree it's a case-by-case thing, but I'm not sure we should be treating `settings.MIDDLEWARE` as what is expected to be on in an environment -- all kinds of python shenanigans can happen in localsettings. What I meant by a reasonable default was simply that, in the absence of this middleware, I'd want our software to respond 'no, standard requests do not have access to all locations' because the opposite sounds like a security issue.